### PR TITLE
Week 5 assignment

### PR DESCRIPTION
**Pull request by konarssuresh**
This PR contains 
* code change for week 5 assignment

### CODE REVIEW

File: Week-5/asyncOp.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+const getNumber1 = () =>
+  new Promise((resolve, _reject) => {
```
Comment (divjot-singh) [inline Week-5/asyncOp.js:2]: One function with async await and one with generator

File: Week-5/hasDuplicate.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const hasDuplicate = (arr) => {
```
Comment (divjot-singh) [inline Week-5/hasDuplicate.js:1]: You can also do it by checking length of set and array

**Review by divjot-singh — APPROVED** [review]

File: Week-5/asyncOp.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
+const getNumber1 = () =>
```
Comment (divjot-singh) [inline Week-5/asyncOp.js:1]: one function with async await and other function using generator

**Review by divjot-singh — CHANGES_REQUESTED** [review]

File: Week-5/asyncOp.js (hunk 1 above)
Comment (konarssuresh) [reply Week-5/asyncOp.js:2]: made changes as discussed

File: Week-5/hasDuplicate.js (hunk 2 above)
Comment (konarssuresh) [reply Week-5/hasDuplicate.js:1]: made suggested change

File: Week-5/asyncOp.js (hunk 3 above)
Comment (konarssuresh) [reply Week-5/asyncOp.js:1]: made changes as discussed

File: Week-5/asyncOp.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  } catch (e) {
+    console.log("error occured");
+    console.log(e);
+  }
+};
+const usingGenerator = () => {
+  console.log("using generator");
+  const gen = asyncOpGenerator();
+  const firstNumberPromise = gen.next().value;
+  firstNumberPromise
```
Comment (divjot-singh) [inline Week-5/asyncOp.js:30]: you can also use async await within generators. instead of promise.then

**Review by divjot-singh — APPROVED** [review]

**Pull request merged**